src/main/java/org/hl7/davinci/priorauth/UpdateClaimTask.java

```diff
@@ -0,0 +1,122 @@
+                    // Send rest-hook notification...
+                    String endpoint = subscription.getChannel().getEndpoint();
+                    logger.info("SubscriptionHandler::Sending rest-hook notification to " + endpoint);
+                    try {
+                        OkHttpClient client = new OkHttpClient();
+                        okhttp3.Response response = client
+                                .newCall(new Request.Builder().post(RequestBody.create(null, "")).url(endpoint).build())
+                                .execute();
+                        logger.fine("SubscriptionHandler::Response " + response.code());
+                        App.getDB().update(Table.SUBSCRIPTION, Collections.singletonMap("id", subscriptionId),
+                                Collections.singletonMap("status",
+                                        SubscriptionStatus.ACTIVE.getDisplay().toLowerCase()));
+                    } catch (IOException e) {
+                        logger.log(Level.SEVERE, "SubscriptionHandler::IOException in request", e);
+                        App.getDB().update(Table.SUBSCRIPTION, Collections.singletonMap("id", subscriptionId),
+                                Collections.singletonMap("status",
+                                        SubscriptionStatus.ERROR.getDisplay().toLowerCase()));
+                    }
+                } else if (subscriptionType == SubscriptionChannelType.WEBSOCKET) {
+                    // Send websocket notification...
+                    String websocketId = App.getDB().readString(Table.SUBSCRIPTION,
+                            Collections.singletonMap("id", subscriptionId), "websocketId");
+                    if (websocketId != null) {
+                        logger.info("SubscriptionHandler::Sending web-socket notification to " + websocketId);
+                        SubscribeController.sendMessageToUser(websocketId, WebSocketConfig.SUBSCRIBE_USER_NOTIFICATION,
+                                "ping: " + subscriptionId);
+                        App.getDB().update(Table.SUBSCRIPTION, Collections.singletonMap("id", subscriptionId),
+                                Collections.singletonMap("status",
+                                        SubscriptionStatus.ACTIVE.getDisplay().toLowerCase()));
+                    } else {
+                        logger.severe("SubscriptionHandler::Unable to send web-socket notification for subscription "
+                                + subscriptionId
+                                + " because web-socket id is null. Client did not bind a websocket to id");
+                        App.getDB().update(Table.SUBSCRIPTION, Collections.singletonMap("id", subscriptionId),
+                                Collections.singletonMap("status",
+                                        SubscriptionStatus.ERROR.getDisplay().toLowerCase()));
+                    }
+                }
+            });
+        }
+    }
+}
```

Comment:
One question, but why not cleave off the cancel logic as well as the update logic?

Reply:
So Cancel is a direct action on the ClaimEndpoint. Update is not. The "Update" here is not a user submitting an update but scheduling an update to pended Claim. The Cancel logic could be refactored to another class but it would not belong here. I think it is okay to leave it in the ClaimEndpoint as it is just a few functions for actions the user can do on the endpoint